packages/story-editor/src/app/story/effects/useHistoryEntry.js

```diff
@@ -23,28 +23,83 @@ import { useEffect, useRef } from '@web-stories-wp/react';
  * Internal dependencies
  */
 import { useHistory } from '../../history';
+import deleteNestedKeys from '../utils/deleteNestedKeys';
+
+// @todo REMOVE ROTATION ANGLE! For testing only!
+const ELEMENT_PROPS_TO_IGNORE = ['resource.baseColor', 'rotationAngle'];
 
 // Record any change to core variables in history (history will know if it's a replay)
 function useHistoryEntry({ story, current, pages, selection, capabilities }) {
   const {
+    state: { currentEntry },
     actions: { stateToHistory },
   } = useHistory();
 
+  const currentHistoryEntryRef = useRef();
+  useEffect(() => {
+    if (currentEntry) {
+      currentHistoryEntryRef.current = JSON.parse(JSON.stringify(currentEntry));
+    }
+  }, [currentEntry]);
+
   const currentPageIndexRef = useRef();
   const selectedElementIdsRef = useRef();
   useEffect(() => {
     currentPageIndexRef.current = current;
     selectedElementIdsRef.current = selection;
   }, [current, selection]);
 
-  useEffect(() => {
-    stateToHistory({
-      story,
-      current: currentPageIndexRef.current,
-      selection: selectedElementIdsRef.current,
-      pages,
-      capabilities,
+  const deleteKeysFromPages = (list) => {
+    // Create a copy of the list not to influence the original.
+    return JSON.parse(JSON.stringify(list)).map((page) => {
+      const { elements } = page;
+      return {
+        ...page,
+        elements: elements.map((element) =>
+          deleteNestedKeys(element, ELEMENT_PROPS_TO_IGNORE)
+        ),
+      };
```

Comment:
Because we're working on a cloned object, we can actually mutate it, so we can do `page.elements = page.elements.map()`. That would make this a bit faster I think.

Reply:
Oh, you are actually mutating it in place in the `deleteNestedKeys` function anyway. So no need to set anything, you can just do:

```
page.elements.forEach(deleteNestedKeys(ELEMENT_PROPS_TO_IGNORE));
```

And then you can change `deleteNestedKeys` to be _curryable_ as in:

```js
const deleteNestedKeys = paths => object => {};
```

And finally there's no need to have the return in `deleteNestedKeys` as it just mutates the object in place.